### PR TITLE
GOG size calc, Steam game size calc improvements

### PR DESCRIPTION
**Pull request by Jeshibu**
I have verified that:
- [x] These changes work, by building the extension and testing.
- [x] That the changes comply with the [rules](https://github.com/darklinkpower/PlayniteExtensionsCollection#contributing) indicated in the repository.
- [x] Pull request is targeting `master` branch.

### CODE REVIEW

File: source/Generic/GamesSizeCalculator/GamesSizeCalculator.cs
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/GamesSizeCalculator.cs:309]: Is it possible that instead of passing all those settings values one by one, pass the whole settings object instead and handle it in the constructor to keep it cleaner perhaps?

File: source/Generic/GamesSizeCalculator/GamesSizeCalculator.cs
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/GamesSizeCalculator.cs:322]: Use curly braces here please. Perhaps it could be done in one line like this

`ICollection<Game> games = settings.Settings.GetUninstalledGameSizeFromSteam ? PlayniteApi.Database.Games : PlayniteApi.Database.Games.Where(x => x.IsInstalled).ToList()`

File: source/Generic/GamesSizeCalculator/GamesSizeCalculator.csproj
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/GamesSizeCalculator.csproj:39]: As mentioned before, if possible use the `Serialization` class to eliminate the dependency

File: source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml:49]: Add localization strings for new strings please

File: source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettings.cs
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettings.cs:44]: Is this collection modifiable by the user or will it never change? If it never changes then it should be moved to the settings Viewmodel object instead as there wouldn't be a purpose in these being here. If they are modifiable then ignore this

File: source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml:9]: Use a Panel as the main control here please. Either `Grid` or `StackPanel` with a margin of `20` for consistency and place the `TabControl as a child`. Also what is the purpose of `ScrollViewer.VerticalScrollBarVisibility` and the ContenAlignment properties?

File: source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml:70]: The Row shouldn't have a `MaxHeight` because it will cause issues if the text is taller, which can happen in bigger fonts. Imo it should be set to `Auto`

File: source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml:75]: Why does it have a `DockPanel.Dock` property if the parent control is a `Grid` ?

File: source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs:15]: This one can be set as static

File: source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs:23]: Should be a constant. Also properties that are not change later should be set to `readonly`

File: source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs:33]: I'm wondering if instead of setting individual fields from settings, it would be possible to set a settings field with the plugin settings and access the properties directly

File: source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs:89]: Models should be in a separate class dedicated to model(s)

File: source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs:121]: Imo a StringBuilder should be used instead of recreating the string with each addition

File: source/Generic/SteamGameTransferUtility/ViewModels/TransferWindowViewModel.cs
Comment (darklinkpower) [inline source/Generic/SteamGameTransferUtility/ViewModels/TransferWindowViewModel.cs:366]: Should not be changed :(

**Review by darklinkpower — CHANGES_REQUESTED** [review]

File: source/Generic/GamesSizeCalculator/GOG/GogApiClient.cs
Comment (Jeshibu) [reply source/Generic/GamesSizeCalculator/GOG/GogApiClient.cs:49]: Yes, Playnite's deserialization throws exceptions when using them in a unit test.

File: source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml
Comment (Jeshibu) [reply source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml:9]: Why an extra wrapping element rather than a Margin="20" for the TabControl?

The properties are for the second tab, I'll move them there.

File: source/Generic/GamesSizeCalculator/GOG/GogApiClient.cs
Comment (Jeshibu) [reply source/Generic/GamesSizeCalculator/GOG/GogApiClient.cs:49]: Playnite's deserialization requires Playnite to be initialized and is unusable for unit tests.

File: source/Generic/GamesSizeCalculator/GOG/GogApiClient.cs
Comment (Jeshibu) [reply source/Generic/GamesSizeCalculator/GOG/GogApiClient.cs]: converted to inline constant since it's not used elsewhere

File: source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs
Comment (Jeshibu) [reply source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs:15]: Unclear what you mean here

File: source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs
Comment (Jeshibu) [reply source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs:23]: It's an interface member, so it can't be static, and properties can't be readonly in C# 7.3

File: source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml
Comment (Jeshibu) [reply source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml:70]: Because its grandparent is a dockpanel. Does this do nothing or something?

File: source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs
Comment (darklinkpower) [reply source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs:19]: That's true, I didn't notice

File: source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs
Comment (darklinkpower) [reply source/Generic/GamesSizeCalculator/Steam/SteamSizeCalculator.cs:15]: private static ILogger logger = LogManager.GetLogger();

But it can be kept this way too

File: source/Generic/GamesSizeCalculator/GamesSizeCalculator.cs
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/GamesSizeCalculator.cs:321]: What's the purpose of using `ICollection` here?

File: source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/GamesSizeCalculatorSettingsView.xaml:75]: Remove the `DockPanel.Dock` please as they don't do anything

File: source/Generic/GamesSizeCalculator/Localization/en_US.xaml
Comment (darklinkpower) [inline source/Generic/GamesSizeCalculator/Localization/en_US.xaml:23]: In my opinion `Region-words` used in the different localization strings is not clear enough and can be confusing, would it be possible to use something else instead? Something like `Excluded regions by word`. For the tab header, maybe `Regions data import settings` or something like that that is in my opinion more clear and eloquent.

Comment (Jeshibu) [reply source/Generic/GamesSizeCalculator/Localization/en_US.xaml:23]: I prefer to leave that to your judgement after merging, though I think "excluded regions by word" makes it sound more magical than it really is

**Comment by darklinkpower** [comment]
Closed because I don't like you

**Pull request closed without merging**
